bin/cat_consensus_sequences.py

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env python
+
+from Bio import SeqIO
+import click
+
+@click.command()
+@click.option("-s", "--sample-name", default="", help="Sample Name.")
+@click.option("-o", "--output-fasta", default="consensus.fasta", help="Consensus Fasta")
+@click.argument("consensus_sequences", nargs=-1)
+def write_consensus(sample_name, output_fasta, consensus_sequences):
+
+    sequence_list = []
+    for infile in consensus_sequences:
+        fasta_file = SeqIO.parse(open(infile), 'fasta')
+        for record in fasta_file:
+            seqid, sequence = record.id.strip(), record.seq
+            segment_number = seqid.split("_")[-2]
+            sequence_list.append([seqid,segment_number,sequence])
+    sequence_list.sort(key=lambda tup: tup[1])
+    with open(output_fasta, 'w') as outfile:
+        for record in sequence_list:
+            outfile.write(f'>{sample_name+"_"+record[1]}\n{record[2]}\n')
```

Comment:
Doesn't look like `seqid` needs to be saved to `sequence_list`. 

```python
for record in sequence_list:
    outfile.write(f'>{sample_name+"_"+record[1]}\n{record[2]}\n')
```
could be rewritten as with tuple destructuring and just f-strings rather than f-string and `+` string concatenation

```python
for _, segment, seq in sequence_list:
    outfile.write(f'>{sample_name}_{segment}\n{seq}\n')
```

Reply:
This process is to reformat seq id as `SampleName_Seg#` (remove segment name) and sort consensus sequences based on segment number before writing to final consensus
[ERR6359501.consensus.txt](https://github.com/peterk87/nf-iav-illumina/files/8732407/ERR6359501.consensus.txt)

Reply:
Having the segment name in the concatenated consensus sequences file would be useful for when we want to do some ad hoc analysis of the seq data. 

Is the segment name being stripped away to be compatible with the BLAST report creation script? If so, the script should be modified and the segment number as well as name should be reported to make it more informative. 

I could also see it being useful to include the top reference information in the sequence comment/description:

```
>{id} {description}
e.g.
>Sample1_1_PB2 MH201221 Influenza A virus (A/PR/8_RGCDC-4,6/34(H1N1)) segment 1 polymerase PB2 (PB2) gene, complete cds
```

Reply:
Segment name stripped away is not compatible with BLAST report script. However for simplicity, I added code to include segment name beside segment number for output report (just to make sure it is work for final consensus and IRMA consensus because IRMA consensus does not contain segment name).